src/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -562,13 +562,13 @@ private void MoveNext(bool isThreadPool)
                 }
                 else
                 {
-                    if (isThreadPool)
+                    if (threadPoolThread is null)
```

Comment:
Is `is null` better than `== null`? That would seem odd to me. (similarly elsewhere)

Reply:
C# 7 ism; `== null` will call the Equals overload if its defined, `is null` always does `ReferenceEquals(...,, null)`

Reply:
I see